paula-pineros/tarea4/calculator.js

```diff
@@ -0,0 +1,125 @@
+//Add the button to the input
+function writeInput(bef) {
+  let expression = document.getElementById("Numero").value;
+  expression += bef;
+  document.getElementById("Numero").value = expression;
+}
+//Clean de input
+function deleteC() {
+  document.getElementById("Numero").value = "";
+}
+//Calculate the exponential
+function exp(a, b) {
+  return a ** b;
+}
+//Calculate the modulo
+function mod(a, b) {
+  return a % b;
+}
+//Calculate the divition
+function div(a, b) {
+  return a / b;
+}
+//Calculate the multiplication
+function mult(a, b) {
+  return a * b;
+}
+//Calculate the substraction
+function sub(a, b) {
+  return a - b;
+}
+//Calculate the add
+function add(a, b) {
+  return a + b;
+}
+
+function splitBySeparators(expression, separators) {
+  let temporal = separators[0];
+  for (let i = 1; i < separators.length; i++) {
+    expression = expression.split(separators[i]).join(temporal);
+  }
+  expression = expression.split(temporal);
+  return expression;
+}
+
+function splitFinal() {
+  let expression = document.getElementById("Numero").value;
+  let separators = ["^", "%", "/", "X", "-", "+"];
+  let expressionInput = document.getElementById("Numero").value;
+  let expressionSplit = splitBySeparators(expression, separators);
+  let size = expressionSplit.length;
+  let len = 0;
+  let expressionFull = [];
+  for (let i = 0; i < size; i++) {
+    expressionSplit[i] = parseFloat(expressionSplit[i]);
+    expressionFull.push(expressionSplit[i]);
+    len += ("" + expressionSplit[i]).length;
+    expressionFull.push(expressionInput[len]);
+    len += 1;
+  }
+  return expressionFull;
+}
+
+//Calculate the total
+function total() {
+  let pieces = splitFinal();
+  pieces.pop();
+  let temporal;
+  let size = pieces.length;
+  let i;
+  for (i = 0; i < size; i++) {
+    if (pieces[i] == "^") {
+      temporal = exp(pieces[i - 1], pieces[i + 1]);
+      pieces[i - 1] = temporal;
+      pieces.splice(i, 2);
+      i--;
+    }
+  }
+  size = pieces.length;
```

Comment:
@PaulaPineros , looks to me like these blocs are repeated, try move to a function to make it reusable